rosserial_python/src/rosserial_python/SerialClient.py

```diff
@@ -493,6 +493,8 @@ def run(self):
                         self.callbacks[topic_id](msg)
                     except KeyError:
                         rospy.logerr("Tried to publish before configured, topic id %d" % topic_id)
+                        rospy.logerr("Exiting...")
+                        rospy.signal_shutdown("Topical callback KeyError")
```

Comment:
This is a massive behaviour change. Please remove from this PR and create a new one if you want to discuss further.